edb/pgsql/parser/ast_builder.py

```diff
@@ -807,8 +813,11 @@ def _build_a_expr(n: Node, c: Context) -> pgast.BaseExpr:
 
 def _build_func_call(n: Node, c: Context) -> pgast.FuncCall:
     n = _unwrap(n, "FuncCall")
+    name = tuple(_list(n, c, "funcname", _build_str))  # type: tuple[str, ...]
+    if name == ('set_config',) or name == ('pg_catalog', 'set_config'):
+        raise PSqlUnsupportedError(n)
     return pgast.FuncCall(
-        name=tuple(_list(n, c, "funcname", _build_str)),
+        name=name,
```

Comment:
This is not really the best place to throw this error.

Consider moving this to here: https://github.com/edgedb/edgedb/blob/master/edb/pgsql/resolver/expr.py#L263-L265